linkerd/app/outbound/src/lib.rs

```diff
@@ -518,17 +518,23 @@ impl Config {
             .push_map_target(|a: listen::Addrs| a.target_addr())
             .into_inner();
 
-        let http = http::DetectHttp::new(
+        const DETECT_BUFFER_CAPACITY: usize = 8192;
+        let http = svc::stack(http::DetectHttp::new(
             h2_settings,
-            detect_protocol_timeout,
             http_server,
             tcp_balance,
             drain.clone(),
-        );
+        ))
+        .push_on_response(transport::Prefix::layer(
+            DETECT_BUFFER_CAPACITY,
+            detect_protocol_timeout,
+        ))
```

Comment:
just a thought: it seems like it might be possible to split the timeout out of the prefix layer and just build a little stack with a prefix layer and then a timeout layer here. i don't know if this is actually worth it, as the code for adding the timeout in the prefix layer is quite simple...so it probably doesn't matter...

Reply:
writing that timeout layer is a bit complicated, as it has to be both inside and outside of the inner service (we're timing out the time between the request entering the outer layer and entering the inner layer). i'm not opposed to it, but i'd rather keep it simple.